### PR TITLE
backends: first bits of kernel-module security backend

### PR DESCRIPTION
**Pull request by stolowski**
This is the first step towards a new backend which maintains kernel modules for some interfaces such as firewall - see https://bugs.launchpad.net/snappy/+bug/1583514? It doesn't do antyhing useful yet.

### CODE REVIEW

File: interfaces/backends/backends.go [hunk 1]
```diff
@@ -35,6 +36,7 @@ var All = []interfaces.SecurityBackend{
 	&dbus.Backend{},
 	&udev.Backend{},
 	&mount.Backend{},
+	&kernel_module.Backend{},
```
Comment (zyga) [inline interfaces/backends/backends.go]: Sorry for nitpicking, could we name this just `module`?

File: interfaces/backends/backends.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/snapcore/snapd/interfaces"
 	"github.com/snapcore/snapd/interfaces/apparmor"
 	"github.com/snapcore/snapd/interfaces/dbus"
+	"github.com/snapcore/snapd/interfaces/kernel_module"
```
Comment (mvo5) [inline interfaces/backends/backends.go]: Sorry for entering bike-shed territory right away. I don't like the "kernel_modules" name, it does not feel go-ish. Maybe `kmod`? But probably something for Gustavo as he is really good with names, OTOH we can keep it for now and change it later when he is back.

**Comment by stolowski** [comment]
Renamed to kmod / SecurityKMod.

**Comment by mvo5** [comment]
Looks good! :+1: once tests are green.

**Comment by zyga** [comment]
Yep, let's merge it.

As the next branch please document the `kmod.Backend` on how it would behave and draft some functions in the `kmod` package itself.

**Pull request merged**